modules/common/util/errors.go

```diff
@@ -23,4 +23,18 @@ var (
 	ErrMoreThanOne = errors.New("Only one should exist")
 	// ErrNoPodSubdomain indicates that there is no Subdomain or Hostname
 	ErrNoPodSubdomain = errors.New("No Subdomain or Hostname")
+	// ErrCreateFailed indicates that the resource failed to create
+	ErrCreateFailed = errors.New("failed to create")
```

Comment:
In place where error is used imo error from serviceAnnotations, err := nad.EnsureNetworksAnnotation(nadList) can be used at this should be enought